server/src/main/java/io/crate/planner/optimizer/tracer/NoOpOptimizerProgressTracker.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.optimizer.tracer;
+
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.planner.optimizer.Rule;
+import io.crate.planner.optimizer.costs.PlanStats;
+
+public class NoOpOptimizerProgressTracker implements OptimizerProgressTracker {
```

Comment:
Could also inline this into the interface, similar to https://github.com/piotrrzysko/crate/blob/cb0d0c57e7b28c7e598b098f6f69573c633e8769/libs/dex/src/main/java/io/crate/data/breaker/RamAccounting.java#L31

Nothing wrong with this and also fine to keep as is, but might make the usages a bit shorter. E.g. `ProgressTracer.NOOP` instead of `NoOpOptimizerProgressTracker.INSTANCE`
(See also comment further below about the naming)

Reply:
Done.